File: SWIG/Examples/test-suite/java/director_basic_runme.java

import director_basic.*;

public class director_basic_runme {

  static {
    try {
      System.loadLibrary("director_basic");
    } catch (UnsatisfiedLinkError e) {
      System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e);
      System.exit(1);
    }
  }

  public static void main(String argv[]) {

      MyFoo a = new MyFoo();

      if (!a.ping().equals("MyFoo::ping()")) {
          throw new RuntimeException ( "a.ping()" );
      }

      if (!a.pong().equals("Foo::pong();MyFoo::ping()")) {
          throw new RuntimeException ( "a.pong()" );
      }

      Foo b = new Foo();

      if (!b.ping().equals("Foo::ping()")) {
          throw new RuntimeException ( "b.ping()" );
      }

      if (!b.pong().equals("Foo::pong();Foo::ping()")) {
          throw new RuntimeException ( "b.pong()" );
      }

      A1 a1 = new A1(1, false);
      a1.delete();
  }
}

class MyFoo extends Foo {
    public String ping() {
        return "MyFoo::ping()";
    }
}

File: Examples/test-suite/kotlin/director_basic_runme.kt
@file:JvmName("director_basic_runme")

import director_basic.*

class KotlinMyFoo : Foo() {
    override fun ping(): String {
        return "KotlinMyFoo::ping()"
    }
}

fun main() {
    try {
        System.loadLibrary("director_basic")
    } catch (e: UnsatisfiedLinkError) {
        System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e)
        kotlin.system.exitProcess(1)
    }

    val a = KotlinMyFoo()

    if (a.ping() != "KotlinMyFoo::ping()") {
        throw RuntimeException("a.ping(): " + a.ping())
    }

    if (a.pong() != "Foo::pong();KotlinMyFoo::ping()") {
        throw RuntimeException("a.pong(): " + a.pong())
    }

    val b = Foo()

    if (b.ping() != "Foo::ping()") {
        throw RuntimeException("b.ping(): " + b.ping())
    }

    if (b.pong() != "Foo::pong();Foo::ping()") {
        throw RuntimeException("b.pong(): " + b.pong())
    }

    val a1 = A1(1, false)
    a1.delete()
}
